### PR TITLE
Migration to yew 0.18.0

### PR DESCRIPTION
**Pull request by Pscheidl**
Makes the components usable with [Yew 0.18.0](https://github.com/yewstack/yew/releases/tag/0.18.0)

## release workflow
We follow [semver](https://semver.org/spec/v2.0.0.html) for versioning this system.

- [x] update `Cargo.toml` `version`.
- [ ] ensure CI completes successfully.
- [ ] add a new tag to the repo matching the new `Cargo.toml` `version`. Either via `git tag` or via the Github UI.
    - all release tags should start with the letter `v` followed by a semver version.
- [ ] CI is configured for release tags and will create a new Github release.
- [ ] once CI for the tag has finished successfully, update the new release page with details on the changes made, which should reflect the content of the `CHANGELOG.md`.

### CODE REVIEW

**Comment by ppiotr3k** [comment]
:question:  @thedodd anything preventing from launching the CI workflow and moving forward on this PR? :pray:

Not being able to work using Yew 0.18 is personally a great pain, especially as [Yew Documentation](https://yew.rs/) only addresses 0.18 now, that Yew Router has been greatly improved to work with it, etc.

**Comment by thedodd** [comment]
@ppiotr3k hey there. Just as a quick note to address timing, you don't actually need to wait for an official release to go out. You can have your Cargo.toml point to the forked source. That way you can immediately start using the updated code.

**Comment by Pscheidl** [comment]
@ppiotr3k 
`ybc = { git = "https://github.com/Pscheidl/ybc/", branch= "yew-0.18.0" }`

**Comment by Pscheidl** [comment]
@thedodd Clippy righteously complained about duplicated `clone`s. Corrected.

**Comment by ppiotr3k** [comment]
@thedodd @Pscheidl thanks for both your replies :bow: 

Actually I figured out there are uncovered changes from moving to Yew 0.18 in @Pscheidl's branch - which in all cases helps moving forward in upgrading to Yew 0.18. I therefore started PR #21  in order start having some tests, with the idea to ease on the upgrade path.

While #21 just proposes a starting point, leveraging existing Rustdoc examples for tests, it already shows IMHO a few other required fixes due to changes in Yew 0.18. I have already started more Rustdoc examples, to both increase documentation and provide test material, but I'm awaiting feedback on #21 to ensure this is an effort worth being taken :slightly_smiling_face:

**Comment by Pscheidl** [comment]
@thedodd Waiting for CI to be ran. Thank you.

**Comment by Follpvosten** [comment]
What's going on here? I'd like to upgrade to 0.18 soon and I don't really want a git dependency or my own fork (although I'm considering it)

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "ybc"
-version = "0.1.6"
+version = "0.1.7"
```
Comment (thedodd) [inline Cargo.toml]: Let's bump this to `0.2.0` as it is a breaking set of changes both for the API as well as the dependencies.

File: src/form/select.rs [hunk 2]
```diff
@@ -155,19 +155,21 @@ impl Component for MultiSelect {
...
             classes.push("is-loading");
         }
+
+        let size: Option<String> = self.props.size.as_ref().map(|size| size.to_string());
         html! {
             <div class=classes>
                 <select
                     multiple=true
-                    size=self.props.list_size
+                    size=size
```
Comment (thedodd) [inline src/form/select.rs]: Looks like this should still reference `list_size`, not `size`. Size is interpolated into the classes, `list_size` is used to control the size of the multi-select list.

**Review by thedodd — CHANGES_REQUESTED** [review]
@Pscheidl thank you for the PR! Sorry that it has taken me so long to review it. Just a few small items to change/discuss.

File: Cargo.toml (hunk 1 above)
Comment (Pscheidl) [reply Cargo.toml]: Absolutely sir, done in https://github.com/thedodd/ybc/pull/20/commits/ed0d26dcc0a94633d57f111a1bc80884fd1caf18

File: src/form/select.rs (hunk 2 above)
Comment (Pscheidl) [reply src/form/select.rs]: Yes, that's correct. Roger that, done in https://github.com/thedodd/ybc/pull/20/commits/bcd6bf395f085ff3641415fd69ca88be8a422041.

**Review by thedodd — APPROVED** [review]
Everything looks great! Looks like CI is failing due to rustfmt lints being issued. This is quite likely due to rust/cargo version updates since the last release. You should be able to automatically incorporate such changes by running rustfmt exactly as CI does (it uses the rustfmt.toml in this repo).

Once that is done and CI is green, we'll be ready to merge this PR and I'll cut a new release.

**Comment by thedodd** [comment]
Disregard, I'll address the linting issues separately `:)`. Cheers mate!

**Comment by Pscheidl** [comment]
Ok, thank you @thedodd. Yup, that was it. I didn't check those details to be honest. I'm on 1.53 and I just use rustfmt automatically.

Thanks you.

**Pull request merged**